test/jdk/java/net/InetAddress/OfLiteralTest.java

```diff
@@ -170,7 +201,63 @@ private static Stream<Arguments> validLiteralArguments() throws Exception {
                 //      with leading 0 that is discarded and address
                 //      parsed as decimal
                 Arguments.of(InetAddressClass.INET_ADDRESS,
-                        "03735928559", ipv4ExpBytes)
+                        "03735928559", ipv4ExpBytes),
+                //      form:'x.x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0 treated as octal segment prefix
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0127.0.0.1", ipv4_87_0_0_1),
+                //      form:'x.x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0 treated as octal segment prefix
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0177.0.0.1", ipv4_127_0_0_1),
+                //      form:'x.x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0s treated as octal segment prefixes
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0177.0000.0000.0001", ipv4_127_0_0_1),
+                //      form:'x.x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0 treated as octal segment prefix
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "127.010.0.1", ipv4_127_8_0_1),
+                //      form:'x.x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0 treated as octal segment prefix
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0377.0377.0377.0377", ipv4_255_255_255_255),
+                //      form:'x.x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0x treated as hexadecimal segment prefixes
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0x7F.0.0x101", ipv4_127_0_1_1),
+                //      form:'x.x' method:InetAddress.ofPosixLiteral -
+                //      with leading 0s treated as octal segment prefixes
+                Arguments.of(InetAddressClass.INET4_ADDRESS_POSIX,
+                        "0177.0200401", ipv4_127_1_1_1),
```

Comment:
Instead of 127.0.1.1 and 127.1.1.1 I would suggest instead testing 127.0.1.2 and 127.1.2.3 so that endianness is demonstrated (i.e. which of 0x102 vs 0x201 gives that result, similarly for octal).

Reply:
done